pkg/server/handler.go

```diff
@@ -225,6 +225,15 @@ func WithWorkspaceProjection(apiHandler http.Handler) http.HandlerFunc {
 			newPath := path.Join("/services/workspaces", cluster.Name.String(), req.URL.Path)
 			klog.V(4).Infof("Rewriting %s -> %s", path.Join(cluster.Name.Path(), req.URL.Path), newPath)
 			req.URL.Path = newPath
+			if shardVirtualWorkspaceURL != nil {
```

Comment:
okay, if the URL wasn't provided we assume the virtual server is in the same process.

Reply:
yes